tests/test_main.py

```diff
@@ -170,11 +170,19 @@ def test_metadata_loads_egg_info(self):
         assert meta['Description'] == 'pôrˈtend'
 
 
-class DiscoveryTests(fixtures.EggInfoPkg, fixtures.DistInfoPkg, unittest.TestCase):
+class DiscoveryTests(
+    fixtures.EggInfoPkg,
+    fixtures.EggInfoPkgPipInstalledNoModules,
+    fixtures.EggInfoPkgSourcesFallback,
+    fixtures.DistInfoPkg,
+    unittest.TestCase,
+):
     def test_package_discovery(self):
         dists = list(distributions())
         assert all(isinstance(dist, Distribution) for dist in dists)
         assert any(dist.metadata['Name'] == 'egginfo-pkg' for dist in dists)
+        assert any(dist.metadata['Name'] == 'empty_egg-pkg' for dist in dists)
+        assert any(dist.metadata['Name'] == 'starved_egg-pkg' for dist in dists)
```

Comment:
Looking at the diff, I find it a little difficult to relate "PipInstalledNoModules" to "empty_egg" and "SourcesFallback" to "starved_egg". Is there a naming that would reflect the relationship more directly?

Reply:
Will rename them to "egg_with_module" and "sources_fallback"

Reply:
Renamed in https://github.com/python/importlib_metadata/pull/437/commits/f62bf95d921fd8aca553813533f110d9010c2412 and https://github.com/python/importlib_metadata/pull/437/commits/a2dc88a299d6cb83052d91928b12cde0a6e5a2de